example/src/main/java/org/xmtp/android/example/pushnotifications/PushNotificationsService.kt

```diff
@@ -0,0 +1,101 @@
+package org.xmtp.android.example.pushnotifications
+
+import android.accounts.AccountManager
+import android.app.PendingIntent
+import android.util.Base64
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.xmtp.android.example.ClientManager
+import org.xmtp.android.example.R
+import org.xmtp.android.example.conversation.ConversationDetailActivity
+import org.xmtp.android.example.extension.truncatedAddress
+import org.xmtp.android.library.messages.EnvelopeBuilder
+import java.util.Date
+
+class PushNotificationsService : FirebaseMessagingService() {
+
+    companion object {
+        private const val TAG = "PushNotificationService"
+
+        internal const val CHANNEL_ID = "xmtp_message"
+    }
+
+    override fun onNewToken(token: String) {
+        super.onNewToken(token)
+        PushNotificationTokenManager.syncPushNotificationsToken(token)
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
```

Comment:
So, the more advanced example is also going to want to subscribe to the `intro` and `invite` topics and handle those messages by refreshing the conversation list and updating the subscriptions. Otherwise notifications are only going to work for pre-existing conversations but not new ones.